src/script/common/c_internal.cpp

```diff
@@ -155,24 +155,28 @@ static void script_log(lua_State *L, const std::string &message,
 		infostream << script_get_backtrace(L) << std::endl;
 }
 
-void log_deprecated(lua_State *L, const std::string &message, int stack_depth)
-{
+DeprecatedHandlingMode getDeprecatedLuaAPIHandlingMode() {
 	static thread_local bool configured = false;
-	static thread_local bool do_log = false;
-	static thread_local bool do_error = false;
+	static thread_local DeprecatedHandlingMode ret = DeprecatedHandlingMode::Ignore;
```

Comment:
I copied this from existing code, but I feel like it's premature optimisation

Reply:
I don't see any improvement by doing this change. At least the code still works.